backend-tests/backend-tests.hs

```diff
@@ -468,6 +470,40 @@ validateHttpResponse cid asset resp = do
 
     _ -> lift $ assertFailure $ "Could not parse header: " <> show h
 
+validateSecurityHeaders :: HasCallStack => HttpResponse -> M ()
```

Comment:
A general question: Did you check that the headers arrive in the browser also through the boundary nodes nginx? In the worst case some of the headers would be removed by some of the proxies and not arrive in the browser end to end. 

This could be checked using a testnet or we just merge this and then check on prod. Ideally we could write a test, but I'm not sure we can do that, as I don't think we have tests going through the boundary nodes, right?

Reply:
If they are not there, then this would be **highly** misleading to developers and imho a bug in the platform.
We know that nginx overwrites certain headers. But this will be resolved later, when we loosen the nginx config (after we have certified headers).
I'm pretty sure other headers are passed through, otherwise the gzip encoding header would also not make it to the client (and this works in production).

Reply:
Sounds good, thanks! Maybe let's still quickly check once this is rolled out, just to be sure ;-)